File: 1774/Roteadores.py
max_int = 2147483647

V, W = [int(x) for x in input().split()]

adj = {}
for i in range(1, V+1):
    adj[i] = []



for i in range(0, W):
    a, b, c = [int(x) for x in input().split()]
    adj[a].append([b,c])
    adj[b].append([a,c])

key = []
mst = []
parent = []

for i in range(0, V+1):
    key.append(max_int)
    mst.append(False)
    parent.append(-1)

key[1] = 0
parent[1] = -1

for i in range(1, V):

    mini = max_int
    u = None

    for v in range(1, V+1):
        if mst[v] == False and key[v] < mini:
            u = v
            mini = key[v]
    

    mst[u] = True

    for it in adj[u]:
        v = it[0]
        w = it[1]
        #print(str(v) + " " +str(w))
        if mst[v] == False and w < key[v]:
            parent[v] = u
            key[v] = w


sum = 0
for i in range(2, V+1):
    sum+=key[i]

print(sum)

File: 1774/Roteadores.cpp
#include <bits/stdc++.h>
using namespace std;

int main (){

    int V, W;
    cin >> V  >> W;

    unordered_map<int, list<pair<int,int>>> adj;

    for (int i = 0; i < W; i++){
        int a, b, c;
        cin >> a >> b >> c;

        adj[a].push_back(make_pair(b,c));
        adj[b].push_back(make_pair(a,c));

    }

    vector<int> key(V+1,INT_MAX);
    vector<bool> mst(V+1, false);
    vector<int> parent(V+1, -1);

    key[1] = 0;
    parent[1] = -1;

    for(int i = 1; i < V; i++){

        int mini = INT_MAX;
        int u;
        //cout << "i " << i << endl;

        for (int v = 1; v <= V; v++){
            //cout << "v " << v << endl;
            if (mst[v] == false && key[v] < mini){
                u = v;
                mini = key[v];
                //cout << "u " << u << endl;
                //cout << "mini " << mini <<endl;
            }
        }

        mst[u] = true;

        for (auto it:adj[u]){
            int v = it.first;
            int w = it.second;
            //cout << "v, w " << v << " " << w << endl;
            //cout << "key[v] " << key[v] << endl;
            //cout << "mst[v] " << mst[v] << endl;
            if (mst[v] == false && w < key[v]){
                parent[v] = u;
                key[v] = w;
            }
        }

    

    }
    int sum = 0;
    for (int i = 2; i<=V; i++){
        sum+=key[i];
    }
    cout << sum << endl;
    

    /*
    for (auto const pair:adj){
        cout << pair.first << endl;
       for (auto const list:pair.second){
        cout << list.first << " " << list.second << endl;
       }
    }
    */
}
